main.go

```diff
@@ -31,13 +31,30 @@ func main() {
 		glog.Info("Built from git commit: ", commit)
 	}
 
-    // Done channel for waiting
-    // fetchPolicyReports := make(chan types.PolicyInfo)
+	// Done channel for waiting
+	// fetchPolicyReports := make(chan types.PolicyInfo)
 
 	monitor := monitor.NewClusterMonitor()
 	go monitor.WatchClusters()
 
-	retriever.NewRetriever(config.Cfg.CCXServer, nil, 2*time.Second, "")
+	ret := retriever.NewRetriever(config.Cfg.CCXServer+"/clusters/reports",
+		config.Cfg.CCXServer+"/content", nil, 2*time.Second, "")
+	//Wait for hub cluster id to make GET API call
+	hubId := "-1"
+	for hubId == "-1" {
+		hubId = monitor.GetLocalCluster()
+		glog.Info("Waiting for local-cluster Id.")
+		time.Sleep(2 * time.Second)
```

Comment:
- Here you should wait only if `hubId` is `-1` and retry until you have a valid hubId.
- It feels like there's a chance for execution to get stuck in this loop.

Reply:
We cannot do anything if the contents are not present .. We will not be able to create any policy without that data . So its okay to  execution to wait , until it got the data . @jlpadilla

Reply:
ok, just wanted to confirm. I see this logic is specific to local-cluster, but maybe it can become a problem if we are processing multiple managed clusters?